wg-meetings/2016-06-01.md

```diff
@@ -0,0 +1,91 @@
+# Node.js Tracing/Diag WG Meeting
+
+- Date: 2016-06-01
+- YouTube: http://www.youtube.com/watch?v=tNxYDvQ1kT8
+
+## Attendees
+
+- @joshgav
+- @mhdawson
+- @matthewloring
+- @ofrobots
+- @paulirish
+- @pavelfeldman
+- @caseq
+- @jasonlaster
+- @pmuellr
+- @fmeawad
++ several more via YouTube
+   
+## Agenda
+
+- TRACE_EVENT - Google's plans and how Node can build on them (@ofrobots)
+- Trace module - @joshgav's proposal - why and what.
+    - [nodejs/tracing-wg#50], [joshgav/node-trace]
+- Chrome Debugging Protocol and v8_inspector (@ofrobots, Google)
+    - [nodejs/node#6792], [nodejs/node#7072]
+    - What it brings to Node and the community now and in the future.
+    - How can the Diagnostics WG utilize it?
+- AsyncWrap [deferred]
+- Expanding scope of this WG [deferred]
+
+## Next Meeting
+
+- Return to monthly cadence - first Wednesday of each month at 12pm US Pacific (immediately before CTC meeting). Next meeting will be July 6.
+
+## Minutes
+
+### Tracing
+
+In Chromium, the Trace Event trace system is implemented through a Tracing Controller. @ofrobots and @matthewloring are working on porting this implementation to V8/D8.
+
+In this Tracing Controller, traces are not streamed, they are collected in memory and dumped on request. It would be up to Node to determine how to flush, stream, etc. in Node.
+
+chrome://tracing is intended for platform developers while “Timeline” view in Chrome DevTools is intended for web/app developers. Data source is the same but filtered differently for each tool.
+
+Instead of “structural tracing” through explicit trace points in code, Google prefers sampling all code. Although slightly less exact on timings this provides more thorough perspective of operations. However, Google hasn’t yet integrated this sampling-based tracing into the default trace configuration.
+
+CDP allows configuration of tracing through the Tracing domain:
+https://chromedevtools.github.io/debugger-protocol-viewer/tot/Tracing/
+
+Google recommends we use the same tracing components but configure appropriately for Node.
+
+Trace output can be processed and visualized by other tools too. Google is evaluating trace output format, considering binary format perhaps based on ProtoBuf. Currently format is agreed upon by Chromium tools teams and documented [here][Trace Events file format]: https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
+
+Chromium’s tracing system and implementation could be used for Node core as well as native and JS modules.
+
+In V8, console.time, console.timeEnd mark a timestamp and send a trace event. Would need a Node-specific JS implementation using Node-appropriate semantics, possibly in `console`. https://chromium.googlesource.com/chromium/src/+/lkcr/third_party/WebKit/Source/core/inspector/ThreadDebugger.cpp#349
+
+4ms per call when tracing is on (0ms when off) with Chromium Tracing Controller. Categories which are disabled are off (no penalty).
```

Comment:
I think the unit of `4ms` is off, 4 milliseconds sounds too much. Maybe `4μs` is correct..?